casepro/utils/__init__.py

```diff
@@ -59,6 +60,23 @@ def json_encode(data):
     return json.dumps(data, cls=JSONEncoder)
 
 
+def json_decode(data):
+    """
+    Decodes the given JSON as primitives
+    """
+    if isinstance(data, six.binary_type):
+        data = data.decode('utf-8')
```

Comment:
Can the encoding change based on the header of the request?

See: http://stackoverflow.com/questions/708915/detecting-the-character-encoding-of-an-http-post-request